src/server/rdb_load.cc

```diff
@@ -1815,6 +1822,10 @@ void RdbLoader::LoadItemsBuffer(DbIndex db_ind, const ItemsBuf& ib) {
     if (item.expire_ms > 0 && db_cntx.time_now_ms >= item.expire_ms)
       continue;
 
+    auto [fit, _] = db_slice.FindExt(db_cntx, item.key);
+    if (IsValid(fit))
+      db_slice.Del(db_cntx.db_index, fit);
+
     auto [it, added] = db_slice.AddEntry(db_cntx, item.key, std::move(pv), item.expire_ms);
```

Comment:
That's ineffective but its just a temporary stub. 

The issue with AddEntry is that I give up my primevalue by moving it. In theory it should:
- insert or fail
- update if failed

Because looking it up separately would make the hot path a lot slower.

Reply:
I've added AddOrUpdate to DbSlice as a possible solution